### PR TITLE
[SPARK-42313][SQL] Assign name to `_LEGACY_ERROR_TEMP_1152`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to integrate _LEGACY_ERROR_TEMP_1152, "PATH_ALREADY_EXISTS".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*`

### CODE REVIEW

**Comment by itholic** [comment]
Test passed. cc @srielau @MaxGekk Can you take a look when you find some time??

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1185,6 +1185,11 @@
     ],
     "sqlState" : "42805"
   },
+  "OUTPUT_PATH_ALREADY_EXISTS" : {
+    "message" : [
+      "Path <outputPath> already exists."
+    ]
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:1191]:
```suggestion
    ],
    "sqlState" : "42K04"
```

**Review by srielau — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Thanks!

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -1191,6 +1191,12 @@
     ],
     "sqlState" : "42805"
   },
+  "OUTPUT_PATH_ALREADY_EXISTS" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1194]: Does the `OUTPUT` word bring any additional info? Can we name the error class as `PATH_ALREADY_EXISTS`?

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -1191,6 +1191,12 @@
...
+    "message" : [
+      "Path <outputPath> already exists."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1196]: Can we suggest something to users, like setting the save mode to a value  `Overwrite = true`?

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 2 above)
Comment (srielau) [reply core/src/main/resources/error/error-classes.json:1194]: +1

**Comment by itholic** [comment]
Applied the suggestions, thanks!!

**Review by MaxGekk — APPROVED** [review]

**Pull request closed without merging**